src/domain/access_token/service.go

```diff
@@ -0,0 +1,37 @@
+package access_token
+
+import (
+	"strings"
+
+	"github.com/davetweetlive/bookstore_oauth-api/src/utils/errors"
+)
+
+type Repository interface {
+	GetById(string) (*AccessToken, *errors.RestErr)
+}
+
+type Service interface {
+	GetById(string) (*AccessToken, *errors.RestErr)
+}
+
+type service struct {
+	repository Repository
+}
+
+func NewService(repo Repository) Service {
```

Comment:
This new service is has Repository parameter whether in application.go file db.NewRepository() is being called. Take a take a relook.

Reply:
Looking into